mbuild/compound.py

```diff
@@ -1154,8 +1186,268 @@ def _kick(self):
+               Library for High Performance Molecular Simulation."
+               J. Chem. Theor. Comput. 9(1): 461-469. (2013).
+
+
+        If using _energy_minimize_openbabel(), please cite:
+        .. [1] O'Boyle, N.M.; Banck, M.; James, C.A.; Morley, C.;
+               Vandermeersch, T.; Hutchison, G.R. "Open Babel: An open
+               chemical toolbox." (2011) J. Cheminf. 3, 33
+
+        .. [2] Open Babel, version X.X.X http://openbabel.org, (installed
+               Month Year)
+
+        If using the 'MMFF94' force field please also cite the following:
+        .. [3] T.A. Halgren, "Merck molecular force field. I. Basis, form,
+               scope, parameterization, and performance of MMFF94." (1996)
+               J. Comput. Chem. 17, 490-519
+        .. [4] T.A. Halgren, "Merck molecular force field. II. MMFF94 van der
+               Waals and electrostatic parameters for intermolecular
+               interactions." (1996) J. Comput. Chem. 17, 520-552
+        .. [5] T.A. Halgren, "Merck molecular force field. III. Molecular
+               geometries and vibrational frequencies for MMFF94." (1996)
+               J. Comput. Chem. 17, 553-586
+        .. [6] T.A. Halgren and R.B. Nachbar, "Merck molecular force field.
+               IV. Conformational energies and geometries for MMFF94." (1996)
+               J. Comput. Chem. 17, 587-615
+        .. [7] T.A. Halgren, "Merck molecular force field. V. Extension of
+               MMFF94 using experimental data, additional computational data,
+               and empirical rules." (1996) J. Comput. Chem. 17, 616-641
+
+        If using the 'MMFF94s' force field please cite the above along with:
+        .. [8] T.A. Halgren, "MMFF VI. MMFF94s option for energy minimization
+               studies." (1999) J. Comput. Chem. 20, 720-729
+
+        If using the 'UFF' force field please cite the following:
+        .. [3] Rappe, A.K., Casewit, C.J., Colwell, K.S., Goddard, W.A. III,
+               Skiff, W.M. "UFF, a full periodic table force field for
+               molecular mechanics and molecular dynamics simulations." (1992)
+               J. Am. Chem. Soc. 114, 10024-10039
+
+        If using the 'GAFF' force field please cite the following:
+        .. [3] Wang, J., Wolf, R.M., Caldwell, J.W., Kollman, P.A., Case, D.A.
+               "Development and testing of a general AMBER force field" (2004)
+               J. Comput. Chem. 25, 1157-1174
+
+        If using the 'Ghemical' force field please cite the following:
+        .. [3] T. Hassinen and M. Perakyla, "New energy terms for reduced
+               protein models implemented in an off-lattice force field" (2001)
+               J. Comput. Chem. 22, 1229-1242
+
+
+
+        """
+        tmp_dir = tempfile.mkdtemp()
+        original = clone(self)
+        self._kick()
+        self.save(os.path.join(tmp_dir, 'un-minimized.mol2'))
+        extension = os.path.splitext(forcefield)[-1]
```

Comment:
I'm thinking it would be better if we could support the OpenBabel force fields (i.e. 'MMFF94', 'MMFF94s', 'UFF', 'GAFF', and 'Ghemical'), user-defined force field XMLs, *and* force field names recognized by Foyer (i.e. `oplsaa` and `trappe-ua`). Perhaps some logic such as:
``` python
openbabel_ffs = ['MMFF94', 'MMFF94s', 'UFF', 'GAFF', 'Ghemical']
if forcefield in openbabel_ffs:
    self._energy_minimize_openbabel(...)
elif extension == '.xml':
    self._energy_minimize_openmm(..., forcefield_files=forcefield, forcefield_name=None)
else:
    self._energy_minimize_openmm(..., forcefield_files=None, forcefield_name=forcefield)
```
And then within `_energy_minimize_openmm` you could have:
``` python
ff = Forcefield(forcefield_files=forcefield_files, name=forcefield_name)
```

This way we also wouldn't need to carry around the large `oplsaa.xml` file because we could just use `forcefield='oplsaa'` in the unit test to point to the opls file within Foyer.